modules/serverless-kafka-broker-with-isolated-dataplane.adoc

```diff
@@ -0,0 +1,89 @@
+
+When using a Broker class of `Kafka`, the Knative Broker implementation for Apache Kafka uses a shared data plane. This means that the `kafka-broker-receiver` and `kafka-broker-dispatcher` deployments in the `knative-eventing` namespace are used for all Apache Kafka Brokers in the cluster.
+
+However, when using a Broker class of `KafkaNamespaced`, the Apache Kafka broker controller creates a new data plane for each namespace where a broker exists. This data plane is used by all `KafkaNamespaced` brokers in that namespace.
+
+This provides isolation between the data planes, so that the `kafka-broker-receiver` and `kafka-broker-dispatcher` deployments in the user namespace are only used for the broker in that namespace.
+
+[IMPORTANT]
+====
+As a consequence of having separate data planes, this security feature creates more deployments and uses more resources. Unless you have such isolation requirements, it is preferable to use a *regular* Broker with a class of `Kafka`.
+====
+
+To create a `KafkaNamespaced` broker, you must set the `eventing.knative.dev/broker.class` annotation to `KafkaNamespaced`.
+
+.Prerequisites
+
+* The {ServerlessOperatorName}, Knative Eventing, and the `KnativeKafka` custom resource are installed on your {product-title} cluster.
+
+* You have access to an Apache Kafka instance such as link:https://access.redhat.com/documentation/en-us/red_hat_amq/7.6/html/amq_streams_on_openshift_overview/kafka-concepts_str#kafka-concepts-key_str[Red Hat AMQ Streams], and have created a Kafka topic.
+
+* You have created a project, or have access to a project with the appropriate roles and permissions to create applications and other workloads in {product-title}.
+
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Create an Apache Kafka-based broker by using a YAML file:
++
+[source,yaml]
+----
+apiVersion: eventing.knative.dev/v1
+kind: Broker
+metadata:
+  annotations:
+    eventing.knative.dev/broker.class: KafkaNamespaced <1>
+  name: default
+  namespace: my-namespace <2>
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: my-config <2>
+    # namespace: my-namespace <3>
+...
+----
+<1> To use the Apache Kafka broker with isolated data planes, the broker class value must be `KafkaNamespaced`.
+<2> The referenced ConfigMap object `my-config` must be in the same namespace as the `Broker` object, in this case `my-namespace`.
+<3> There is no need to define the namespace for the ConfigMap object as it defaults to the Broker's namespace.
+
+. Apply the Apache Kafka-based broker YAML file:
++
+[source,terminal]
+----
+$ oc apply -f <filename>
+----
+
+[IMPORTANT]
+====
+The `ConfigMap` object that is specified in `spec.config` **must** be in the same namespace as the `Broker` object:
```

Comment:
```suggestion
The `ConfigMap` object that is specified in `spec.config` must be in the same namespace as the `Broker` object:
```
- per IBM Style, do not use text formatting like bold, italics, or underlining to indicate emphasis. In this case, the Important admonition already emphasizes the content sufficiently.